docs/server-side-apply.md

```diff
@@ -0,0 +1,232 @@
+# Applying changes to Prometheus resource generated by MonitoringStack using Server Side Apply
+
+
+ ## Prerequisites
+ - Setup a cluster with following MonitoringStack spec
+
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  labels:
+    obo: example
+  name: sample-monitoring-stack
+  namespace: obo-demo
+spec:
+  logLevel: debug
+  retention: 1d
+  resourceSelector:
+    matchLabels:
+      app: demo
+```
+
+This will generate a Prometheus resource named `sample-monitoring-stack` in `obo-demo` namespace.
+
+Get the managed fields of the generated Prometheus resource using command 
+```sh
+oc -n obo-demo get Prometheus.monitoring.rhobs -oyaml --show-managed-fields
+```
+
+```yaml
+managedFields:
+- apiVersion: monitoring.rhobs/v1
+  fieldsType: FieldsV1
+  fieldsV1:
+    f:metadata:
+      f:labels:
+        f:app.kubernetes.io/managed-by: {}
+        f:app.kubernetes.io/name: {}
+        f:app.kubernetes.io/part-of: {}
+      f:ownerReferences:
+        k:{"uid":"81da0d9a-61aa-4df3-affc-71015bcbde5a"}: {}
+    f:spec:
+      f:additionalScrapeConfigs: {}
+      f:affinity:
+        f:podAntiAffinity:
+          f:requiredDuringSchedulingIgnoredDuringExecution: {}
+      f:alerting:
+        f:alertmanagers: {}
+      f:arbitraryFSAccessThroughSMs: {}
+      f:logLevel: {}
+      f:podMetadata:
+        f:labels:
+          f:app.kubernetes.io/component: {}
+          f:app.kubernetes.io/part-of: {}
+      f:podMonitorSelector: {}
+      f:replicas: {}
+      f:resources:
+        f:limits:
+          f:cpu: {}
+          f:memory: {}
+        f:requests:
+          f:cpu: {}
+          f:memory: {}
+      f:retention: {}
+      f:ruleSelector: {}
+      f:rules:
+        f:alert: {}
+      f:securityContext:
+        f:fsGroup: {}
+        f:runAsNonRoot: {}
+        f:runAsUser: {}
+      f:serviceAccountName: {}
+      f:serviceMonitorSelector: {}
+      f:thanos:
+        f:baseImage: {}
+        f:resources: {}
+        f:version: {}
+      f:tsdb: {}
+  manager: obo-demo/sample-monitoring-stack
```

Comment:
shouldn't the manager be the controller or the operator and not the actual resource itself ?

Reply:
thats what Obo sets as the manager
https://github.com/rhobs/observability-operator/blob/a977d1554c0bf8de8578e3770a6860bc0ca244a7/pkg/reconciler/reconciler.go#L37-L38

Reply:
can there be more than one `MonitoringStack` managed by Obo?

Reply:
yes, definitely but IIUC, I think we should set the fieldManager to something static that represents the obo instead of the treating it like an ownerReference.